src/navigation/failure_identification/watchdog.py

```diff
@@ -0,0 +1,69 @@
+import numpy as np
+import rospy
+from pandas import DataFrame
+
+# 10 or more data objects flagged as stuck, then you are actually stuck
+DF_THRESHOLD = 100
+STUCK_THRESHOLD = 50
+ANGULAR_THRESHOLD = 0.001
+LINEAR_THRESHOLD = 0.55
+
+
+class WatchDog:
+    def __init__(self, collector_in):
+        self.collector = collector_in
+
+    # def is_stuck(self, dataframe: DataFrame):
+    #     return False
+    def update_pointers(self):
+        self.collector.right_pointer = self.collector.row_counter
+        self.collector.left_pointer = self.collector.right_pointer - DF_THRESHOLD
+        rospy.logerr(f"RIGHT: {self.collector.right_pointer} LEFT: {self.collector.left_pointer}")
+
+    def check_angular_stuck(self, dataframe: DataFrame):
+        counter = 0
+        self.collector.left_pointer = self.collector.right_pointer - DF_THRESHOLD
+        while self.collector.left_pointer < self.collector.right_pointer:
+            actual = (dataframe.loc[self.collector._df["row"] == self.collector.left_pointer]["angular_velocity"]).iloc[
```

Comment:
You shouldn't need to access self.collecter._df right? I think dataframe should be the same object